### ISSUE TITLE
invalid-name match for a valid name in except

### ISSUE THREAD

**Issue by korhojoa** [pylint-dev/pylint#10652]
### Bug description

```python
C0103: Variable name "VERSION" doesn't conform to snake_case naming style (invalid-name)
This reproduces:


from importlib.metadata import PackageNotFoundError
from importlib.metadata import version

try:
    VERSION = version("ty")
except PackageNotFoundError:
    VERSION = "0.0.0"


3.3.9 does not match this.
```

### Configuration

```ini
[tool.pylint.format]
max-line-length = 120

[tool.pylint.basic]
good-names = ["i", "j", "k", "ex", "Run", "_", "f", "dn"]
extension-pkg-whitelist = ["lxml.etree", "pydantic"]

[tool.pylint."messages control"]
disable = [
    "raw-checker-failed",
    "locally-disabled",
    "suppressed-message",
    "wrong-import-order",
    "ungrouped-imports",
]

[tool.pylint.design]
max-args = 8
max-positional-arguments = 8
max-attributes = 10
min-public-methods = 0
```

### Command used

```shell
pylint a
```

### Pylint output

```python
************* Module a
a/__init__.py:9:4: C0103: Variable name "VERSION" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

No output.

### Pylint version

```shell
pylint 4.0.0
astroid 4.0.1
Python 3.11.12 (main, May  8 2025, 08:02:37) [GCC 13.3.0]
```

### Additional dependencies

```python

```

**Comment by jacobtylerwalls** [comment]
Thanks, looks like relying on `astroid.nodes.are_exclusive` is not sufficient here. I tried moving the first assignment to an `else` block and still had `invalid-name`, which I expected `are_exclusive` to handle!

**Comment by nmlorg** [comment]
I think this is what I just ran into, and it seems to be a combination of a generated value and any indentation:&nbsp; The example from the release announcement triggers the problem if you replace either number with a function call (like `importlib.metadata.version`) or object instantiation:

```
$ cat aaa.py
class Class: pass
$ cat test.py
"""Test."""

import aaa

AAA = aaa.Class()

SOMETIMES = True

if SOMETIMES:
    LIMIT = 1
else:
    LIMIT = 500

if SOMETIMES:
    BBB = aaa.Class()
else:
    BBB = 500
$ pylint --version
pylint 4.0.0
astroid 4.0.1
Python 3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0]
$ pylint test.py
************* Module test
test.py:15:4: C0103: Variable name "BBB" doesn't conform to snake_case naming style (invalid-name)
test.py:17:4: C0103: Variable name "BBB" doesn't conform to snake_case naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.78/10 (previous run: 7.78/10, +0.00)

$ pip install -U pylint==3.3.9
$ pylint --version
pylint 3.3.9
astroid 3.3.11
Python 3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0]
$ pylint test.py

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 7.78/10, +2.22)
```

**Comment by MattBuckley-TfN** [comment]
I have found a similar issue the following code, which when setting module level constants from the result of a function, emits invalid-name (C0103) from pylint. However, C0103 is not emitted when TEST is set to a dictionary directly instead of calling the function.

<details><summary>Versions</summary>
 
- pylint 4.0.1
- astroid 4.0.1
- Python 3.13.7 | packaged by conda-forge | (main, Sep  3 2025, 14:18:04) [MSC v.1944 64 bit (AMD64)]
</details>

# Code with C0103
```python
"""Test"""

import sys


def f():
    """Dummy function."""
    return {"test": 1}


if sys.version_info.minor <= 10:
    TEST = f()
else:
    TEST = {"test": 2}
```

## Pylint Result
```
************* Module test
test.py:12:4: C0103: Variable name "TEST" doesn't conform to snake_case naming style (invalid-name)
test.py:14:4: C0103: Variable name "TEST" doesn't conform to snake_case naming style (invalid-name)
```

---

# Code Without C0103

```python
if sys.version_info.minor <= 10:
    TEST = {"test": 1}
else:
    TEST = {"test": 2}
```

**Comment by nmlorg** [comment]
```
$ pip install -U pylint==4.0.1
Successfully installed pylint-4.0.1

$ pylint korhojoa.py
************* Module korhojoa
korhojoa.py:9:4: C0103: Variable name "VERSION" doesn't conform to snake_case naming style (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 10.00/10, -1.67)

$ pylint test.py
************* Module test
test.py:15:4: C0103: Variable name "BBB" doesn't conform to snake_case naming style (invalid-name)
test.py:17:4: C0103: Variable name "BBB" doesn't conform to snake_case naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.78/10 (previous run: 7.78/10, +0.00)

$ pip install -U pylint
Successfully installed pylint-4.0.3

$ pylint korhojoa.py

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.33/10, +1.67)

$ pylint test.py
************* Module test
test.py:15:4: C0103: Variable name "BBB" doesn't conform to snake_case naming style (invalid-name)
test.py:17:4: C0103: Variable name "BBB" doesn't conform to snake_case naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.78/10 (previous run: 7.78/10, +0.00)
```

<br>
I don't have permission to reopen; should I open a separate issue?

**Comment by jacobtylerwalls** [comment]
What is the issue you are trying to report?

**Comment by nmlorg** [comment]
#10678 did not correct:

```py
if SOMETIMES:
    BBB = aaa.Class()
else:
    BBB = 500
```

(see https://github.com/pylint-dev/pylint/issues/10652#issuecomment-3399248789).

**Comment by nmlorg** [comment]
(Moved to #10745.)
